atlp-rwanda/phoenix-bn-backend
# Signup to barefoot using email&password and user verification

#### What does this PR do?
>  Creates the following models
> - `Users`
> - `Roles`
>  Adds the following usefull packages
> - `jsonwebtoken` for generating tokens and verification link
> - `nodemailer` for sending emails(verification emails)
> - `bcrypt` for encrypting the password
> - `joi` used as validator

#### Description of Task to be completed
> - Save new user data into the database
> - Send verification link to provided email
> - Verify user's account and log him/her in

#### How should this be manually tested?
> - Clone this repository into your PC
> - Switch to the `ft-signup-withEmailAndPassword` branch
> - Run command `npm install` to install dependencies packages
> - create `.env` file and create the following variables with it's values basing on your database:
>      - `DB_PORT = 'port number'`
>      - `DB_HOST = 'host'`
>      - `DB_USER = 'database username'`
>      - `DB_PASSWORD = 'database password'`
>      - `DB_NAME = 'database name'`
>      - `DB_DIALECT =' dialect'`
>      - `DATABASE_URL=`
>	 - `JWTKEY=`
>	 - `PORT=`
>	 - `API_VERSION=v1`
> -EMAIL
>-PASSWORD
>
> Run `npm start` to create migrations  ,seeders and starting the application.
> check the port your application runs on
> open postman and send post request to this endpoint 'http://localhost:`port`/api/v1/users/signup' and provide the following fields
![creatingUser](https://user-images.githubusercontent.com/26524106/96674904-0d910100-136a-11eb-9c39-13d0f27a0ddc.png)

> check your inbox to get verification link
>  'because your testing loccaly you have to change the url in your in box to this `http://localhost:port/api/v1/users/verify/token'
> -send a get request using postman or your browser
![verify](https://user-images.githubusercontent.com/26524106/96674922-171a6900-136a-11eb-893a-105bd9e29fbc.png)

#### What are the relevant Trello story?
> [Sign up to Barefoot & Users verification - Manual](https://trello.com/c/3zwZIJe4)

Thank you !!
